### PR TITLE
add test cases for GroupBy.apply trivial cases 

### PR DESCRIPTION
**Pull request by zym1010**
- [x] regarding #20066
- 2 tests added, one should pass, one expected to fail.
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -233,6 +233,39 @@ def test_apply_issues(self):
             lambda x: x['time'][x['value'].idxmax()])
         assert_series_equal(result, expected)
 
+    def test_apply_trivial(self):
+        # GH 20066
+        def gen_one_df(start=0):
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: you dont need a function for this, just create the dataframe directly

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -233,6 +233,39 @@ def test_apply_issues(self):
...
+
+        df = gen_one_df()
+        expected = pd.concat([gen_one_df(1), gen_one_df(1)],
+                             axis=1, keys=['float64', 'object'])
+        result = df.groupby([str(x) for x in df.dtypes],
+                            axis=1).apply(lambda x: gen_one_df(1))
+
+        assert_frame_equal(result, expected)
+
+    @pytest.mark.xfail(reason="didn't work yet")
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: can you mark the issue and provide some more explanation

File: pandas/tests/groupby/test_groupby.py [hunk 3]
```diff
@@ -233,6 +233,39 @@ def test_apply_issues(self):
             lambda x: x['time'][x['value'].idxmax()])
         assert_series_equal(result, expected)
 
+    def test_apply_trivial(self):
+        # GH 20066
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: can you provide a 1-line comment

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_groupby.py (hunk 1 above)
Comment (zym1010) [reply pandas/tests/groupby/test_groupby.py]: done in my last commit.

File: pandas/tests/groupby/test_groupby.py (hunk 3 above)
Comment (zym1010) [reply pandas/tests/groupby/test_groupby.py]: done in my last commit.

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (zym1010) [reply pandas/tests/groupby/test_groupby.py]: done in my last commit.

**Comment by jreback** [comment]
slightly off topic. It would be nice to split up tests_groupby.py somewhat if you would like to tackle this.  Ideally would group some similar tests together and split out some to either existing sub-modules or some new ones.

**Comment by zym1010** [comment]
@jreback probably not right now. I've just learned to use GroupBy stuffs...

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks. will merge after the doc sprint.

**Comment by jreback** [comment]
thanks @zym1010 

love for you to look into this issue (or others!)

**Pull request merged**